influxdb.js

```diff
@@ -1,170 +1,331 @@
 var _ = require('lodash');
 
-module.exports = function(RED) {
+module.exports = function (RED) {
     "use strict";
     var Influx = require('influx');
+    var { InfluxDB, Point } = require('@influxdata/influxdb-client');
 
     /**
-     * Config node.  Currently we only connect to one host.
+     * Config node. Currently we only connect to one host.
      */
     function InfluxConfigNode(n) {
-        RED.nodes.createNode(this,n);
+        RED.nodes.createNode(this, n);
         this.hostname = n.hostname;
         this.port = n.port;
-        this.database= n.database;
+        this.database = n.database;
         this.name = n.name;
-        this.usetls = n.usetls;
-        if (typeof this.usetls === 'undefined') {
-            this.usetls = false;
+
+        var clientOptions = null;
+        if (n.influxdbVersion === '1.8') {
+            this.usetls = n.usetls;
+            if (typeof this.usetls === 'undefined') {
+                this.usetls = false;
+            }
+            // for backward compatibility with old 'protocol' setting
+            if (n.protocol === 'https') {
+                this.usetls = true;
+            }
+            if (this.usetls && n.tls) {
+                var tlsNode = RED.nodes.getNode(n.tls);
+                if (tlsNode) {
+                    this.hostOptions = {};
+                    tlsNode.addTLSOptions(this.hostOptions);
+                }
+            }
+
+            this.client = new Influx.InfluxDB({
+                hosts: [{
+                    host: this.hostname,
+                    port: this.port,
+                    protocol: this.usetls ? "https" : "http",
+                    options: this.hostOptions
+                }],
+                database: this.database,
+                username: this.credentials.username,
+                password: this.credentials.password
+            });
+        } else if (n.influxdbVersion === '1.8 flux') {
+            var token = `${this.credentials.username}:${this.credentials.password}`;
+            clientOptions = {
+                url: n.url,
+                token: token
```

Comment:
would be good to support the `rejectUnauthorized` client option here (set to false) so that it can be configured to use a self-signed certificate without errors, and without the need to set an environment variable on startup.